// te-one-utils/dist/coordtransform.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.wgstobd = exports.getDistance = exports.wgs84tobd09 = exports.bd09towgs84 = exports.gcj02towgs84 = exports.wgs84togcj02 = exports.bd09togcj02 = exports.gcj02tobd09 = exports.CoordinateItem = void 0;
const EARTH_RADIUS = 6378245.0;
const EE = 0.00669342162296594323;
const PI = 3.1415926535897932384626;
const x_PI = (3.14159265358979324 * 3000.0) / 180.0;
const RADIAN = PI / 180.0;
class CoordinateItem {
    constructor() {
        this.longitude = 0;
        this.latitude = 0;
    }
}
exports.CoordinateItem = CoordinateItem;
function gcj02tobd09(gcj_lng, gcj_lat) {
    let bd_lat;
    let bd_lng;
    let x = gcj_lng, y = gcj_lat;
    let z = Math.sqrt(x * x + y * y) + 0.00002 * Math.sin(y * x_PI);
    let theta = Math.atan2(y, x) + 0.000003 * Math.cos(x * x_PI);
    bd_lng = z * Math.cos(theta) + 0.0065;
    bd_lat = z * Math.sin(theta) + 0.006;
    const mapInfo = new CoordinateItem();
    mapInfo.latitude = bd_lat;
    mapInfo.longitude = bd_lng;
    return mapInfo;
}
exports.gcj02tobd09 = gcj02tobd09;
;
function bd09togcj02(bd_lng, bd_lat) {
    let x = bd_lng - 0.0065, y = bd_lat - 0.006;
    let z = Math.sqrt(x * x + y * y) - 0.00002 * Math.sin(y * x_PI);
    let theta = Math.atan2(y, x) - 0.000003 * Math.cos(x * x_PI);
    const mapInfo = new CoordinateItem();
    mapInfo.longitude = z * Math.cos(theta);
    mapInfo.latitude = z * Math.sin(theta);
    return mapInfo;
}
exports.bd09togcj02 = bd09togcj02;
;
function wgs84togcj02(wgs_lng, wgs_lat) {
    const mapInfo = new CoordinateItem();
    if (outOfChina(wgs_lng, wgs_lat)) {
        mapInfo.longitude = wgs_lng;
        mapInfo.latitude = wgs_lat;
    }
    else {
        let dlat = transformlat(wgs_lng - 105.0, wgs_lat - 35.0);
        let dlng = transformlng(wgs_lng - 105.0, wgs_lat - 35.0);
        const radlat = (wgs_lat / 180.0) * PI;
        let magic = Math.sin(radlat);
        magic = 1 - EE * magic * magic;
        const sqrtmagic = Math.sqrt(magic);
        dlat = (dlat * 180.0) / (((EARTH_RADIUS * (1 - EE)) / (magic * sqrtmagic)) * PI);
        dlng = (dlng * 180.0) / ((EARTH_RADIUS / sqrtmagic) * Math.cos(radlat) * PI);
        mapInfo.longitude = wgs_lng + dlng;
        mapInfo.latitude = wgs_lat + dlat;
    }
    return mapInfo;
}
exports.wgs84togcj02 = wgs84togcj02;
function gcj02towgs84(gcj_lng, gcj_lat) {
    const mapInfo = new CoordinateItem();
    if (outOfChina(gcj_lng, gcj_lat)) {
        mapInfo.longitude = gcj_lng;
        mapInfo.latitude = gcj_lat;
    }
    else {
        let dlat = transformlat(gcj_lng - 105.0, gcj_lat - 35.0);
        let dlng = transformlng(gcj_lng - 105.0, gcj_lat - 35.0);
        const radlat = (gcj_lat / 180.0) * PI;
        let magic = Math.sin(radlat);
        magic = 1 - EE * magic * magic;
        const sqrtmagic = Math.sqrt(magic);
        dlat = (dlat * 180.0) / (((EARTH_RADIUS * (1 - EE)) / (magic * sqrtmagic)) * PI);
        dlng = (dlng * 180.0) / ((EARTH_RADIUS / sqrtmagic) * Math.cos(radlat) * PI);
        const mglat = gcj_lat + dlat;
        const mglng = gcj_lng + dlng;
        mapInfo.longitude = gcj_lng * 2 - mglng;
        mapInfo.latitude = gcj_lat * 2 - mglat;
    }
    return mapInfo;
}
exports.gcj02towgs84 = gcj02towgs84;
function bd09towgs84(bd_lng, bd_lat) {
    const mapGCJInfo = bd09togcj02(bd_lng, bd_lat);
    return gcj02towgs84(mapGCJInfo.longitude, mapGCJInfo.latitude);
}
exports.bd09towgs84 = bd09towgs84;
function wgs84tobd09(wgs_lng, wgs_lat) {
    const mapGCJInfo = wgs84togcj02(wgs_lng, wgs_lat);
    return gcj02tobd09(mapGCJInfo.longitude, mapGCJInfo.latitude);
}
exports.wgs84tobd09 = wgs84tobd09;
function getDistance(lng1, lat1, lng2, lat2) {
    let radLat1 = lat1 * RADIAN;
    let radLat2 = lat2 * RADIAN;
    let a = radLat1 - radLat2;
    let b = lng1 * RADIAN - lng2 * RADIAN;
    const s = 2 * Math.asin(Math.sqrt(Math.pow(Math.sin(a / 2), 2) + Math.cos(radLat1) * Math.cos(radLat2) * Math.pow(Math.sin(b / 2), 2)));
    return Math.round(s * EARTH_RADIUS * 10000) / 10000;
}
exports.getDistance = getDistance;
;
function wgstobd(wgs_lng, wgs_lat) {
    const mapInfo = new CoordinateItem();
    if (wgs_lng != 0 && wgs_lat != 0) {
        if (outOfChina(wgs_lng, wgs_lat)) {
            mapInfo.longitude = wgs_lng;
            mapInfo.latitude = wgs_lat;
        }
        else {
            mapInfo.longitude = wgs_lng + 0.0065;
            mapInfo.latitude = wgs_lat + 0.006;
        }
    }
    return mapInfo;
}
exports.wgstobd = wgstobd;
function outOfChina(lng, lat) {
    return !(lng > 73.66 && lng < 135.05 && lat > 3.86 && lat < 53.55);
}
function transformlng(lng, lat) {
    let ret = 300.0 + lng + 2.0 * lat + 0.1 * lng * lng + 0.1 * lng * lat + 0.1 * Math.sqrt(Math.abs(lng));
    ret += (20.0 * Math.sin(6.0 * lng * PI) + 20.0 * Math.sin(2.0 * lng * PI)) * 2.0 / 3.0;
    ret += (20.0 * Math.sin(lng * PI) + 40.0 * Math.sin(lng / 3.0 * PI)) * 2.0 / 3.0;
    ret += (150.0 * Math.sin(lng / 12.0 * PI) + 300.0 * Math.sin(lng / 30.0 * PI)) * 2.0 / 3.0;
    return ret;
}
function transformlat(lng, lat) {
    let ret = -100.0 + 2.0 * lng + 3.0 * lat + 0.2 * lat * lat + 0.1 * lng * lat + 0.2 * Math.sqrt(Math.abs(lng));
    ret += (20.0 * Math.sin(6.0 * lng * PI) + 20.0 * Math.sin(2.0 * lng * PI)) * 2.0 / 3.0;
    ret += (20.0 * Math.sin(lat * PI) + 40.0 * Math.sin(lat / 3.0 * PI)) * 2.0 / 3.0;
    ret += (160.0 * Math.sin(lat / 12.0 * PI) + 320 * Math.sin(lat * PI / 30.0)) * 2.0 / 3.0;
    return ret;
}
