// te-one-utils/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.wgstobd = exports.getDistance = exports.wgs84tobd09 = exports.bd09towgs84 = exports.gcj02towgs84 = exports.wgs84togcj02 = exports.bd09togcj02 = exports.gcj02tobd09 = exports.requireToImport = exports.getDataManager = exports.destroyConsole = exports.isEmptyObject = exports.getRootByFile = exports.reverseMap = exports.isContains = exports.promisify2 = exports.serialize = exports.SymbolKey = exports.promisify = exports.parseXml = exports.eventbus = exports.getType = exports.loading = exports.getHash = exports.unique = exports.clone = exports.shell = exports.mkdir = exports.rmrf = exports.md5 = exports.JParse = exports.decrypt = exports.aes = exports.ip = exports.$ = void 0;
const _1 = require("./$");
exports.$ = _1.default;
const coordtransform_1 = require("./coordtransform");
Object.defineProperty(exports, "gcj02tobd09", { enumerable: true, get: function () { return coordtransform_1.gcj02tobd09; } });
Object.defineProperty(exports, "bd09togcj02", { enumerable: true, get: function () { return coordtransform_1.bd09togcj02; } });
Object.defineProperty(exports, "wgs84togcj02", { enumerable: true, get: function () { return coordtransform_1.wgs84togcj02; } });
Object.defineProperty(exports, "gcj02towgs84", { enumerable: true, get: function () { return coordtransform_1.gcj02towgs84; } });
Object.defineProperty(exports, "bd09towgs84", { enumerable: true, get: function () { return coordtransform_1.bd09towgs84; } });
Object.defineProperty(exports, "wgs84tobd09", { enumerable: true, get: function () { return coordtransform_1.wgs84tobd09; } });
Object.defineProperty(exports, "getDistance", { enumerable: true, get: function () { return coordtransform_1.getDistance; } });
Object.defineProperty(exports, "wgstobd", { enumerable: true, get: function () { return coordtransform_1.wgstobd; } });
const aes_1 = require("./aes");
exports.aes = aes_1.default;
Object.defineProperty(exports, "decrypt", { enumerable: true, get: function () { return aes_1.decrypt; } });
const md5_1 = require("./md5");
exports.md5 = md5_1.default;
const rmrf_1 = require("./rmrf");
exports.rmrf = rmrf_1.default;
const mkdir_1 = require("./mkdir");
exports.mkdir = mkdir_1.default;
const clientip_1 = require("./clientip");
exports.ip = clientip_1.default;
const clone_1 = require("./clone");
exports.clone = clone_1.default;
const shell_1 = require("./shell");
exports.shell = shell_1.default;
const util = require("util");
const unique_1 = require("./unique");
exports.unique = unique_1.default;
const getHash_1 = require("./getHash");
exports.getHash = getHash_1.default;
const loading_1 = require("./loading");
exports.loading = loading_1.default;
const getType_1 = require("./getType");
exports.getType = getType_1.default;
const eventbus_1 = require("./eventbus");
exports.eventbus = eventbus_1.default;
const symbolkey_1 = require("./symbolkey");
exports.SymbolKey = symbolkey_1.default;
const serialize_1 = require("./serialize");
exports.serialize = serialize_1.default;
const safeJSONParse_1 = require("./safeJSONParse");
exports.JParse = safeJSONParse_1.default;
const promisify_1 = require("./promisify");
exports.promisify = promisify_1.default;
const xml2js_1 = require("xml2js");
const promisify2_1 = require("./promisify2");
exports.promisify2 = promisify2_1.default;
const isContains_1 = require("./isContains");
exports.isContains = isContains_1.default;
const reverseMap_1 = require("./reverseMap");
Object.defineProperty(exports, "reverseMap", { enumerable: true, get: function () { return reverseMap_1.reverseMap; } });
const getRootByFile_1 = require("./getRootByFile");
exports.getRootByFile = getRootByFile_1.default;
const isEmptyObject_1 = require("./isEmptyObject");
exports.isEmptyObject = isEmptyObject_1.default;
const getDataManager_1 = require("./getDataManager");
exports.getDataManager = getDataManager_1.default;
const requireToImport_1 = require("./requireToImport");
exports.requireToImport = requireToImport_1.default;
const parseXml = util.promisify(xml2js_1.parseString);
exports.parseXml = parseXml;
function destroyConsole() {
    console = new Proxy({}, {
        get: function () {
            return function () { };
        },
    });
}
exports.destroyConsole = destroyConsole;
exports.default = {
    $: _1.default,
    ip: clientip_1.default,
    aes: aes_1.default,
    md5: md5_1.default,
    rmrf: rmrf_1.default,
    mkdir: mkdir_1.default,
    shell: shell_1.default,
    clone: clone_1.default,
    unique: unique_1.default,
    getHash: getHash_1.default,
    decrypt: aes_1.decrypt,
    JParse: safeJSONParse_1.default,
    loading: loading_1.default,
    getType: getType_1.default,
    parseXml,
    eventbus: eventbus_1.default,
    promisify: promisify_1.default,
    SymbolKey: symbolkey_1.default,
    serialize: serialize_1.default,
    promisify2: promisify2_1.default,
    isContains: isContains_1.default,
    reverseMap: reverseMap_1.reverseMap,
    getRootByFile: getRootByFile_1.default,
    isEmptyObject: isEmptyObject_1.default,
    destroyConsole,
    getDataManager: getDataManager_1.default,
    requireToImport: requireToImport_1.default,
    gcj02tobd09: coordtransform_1.gcj02tobd09,
    bd09togcj02: coordtransform_1.bd09togcj02,
    wgs84togcj02: coordtransform_1.wgs84togcj02,
    gcj02towgs84: coordtransform_1.gcj02towgs84,
    bd09towgs84: coordtransform_1.bd09towgs84,
    wgs84tobd09: coordtransform_1.wgs84tobd09,
    getDistance: coordtransform_1.getDistance,
    wgstobd: coordtransform_1.wgstobd
};
